src/accumulation.jl

```diff
@@ -24,6 +24,18 @@ function add!!(x, t::InplaceableThunk)
     end
 end
 
+function add!!(x, t::BroadcastThunk)
+    return if is_inplaceable_destination(x)
+        if !debug_mode()
+            x .+= t.bc
```

Comment:
We should teach `debug_add!(x::BroadcastedThunk` no?

Reply:
Or delete that call here. It's just checking that `x === x`.